src/gba/regs.nim

```diff
@@ -127,54 +127,47 @@ type
     enable* {.bitsize:1.}: bool
     notUsed2* {.bitsize:8.}: cuint
 
-  KeypadRegs = KEYINPUT | KEYCNT
-
-  KEYINPUT* {.packed.} = object
-    a* {.bitsize:1.}: bool
-    b* {.bitsize:1.}: bool
-    select* {.bitsize:1.}: bool
-    start* {.bitsize:1.}: bool
-    right* {.bitsize:1.}: bool
-    left* {.bitsize:1.}: bool
-    up* {.bitsize:1.}: bool
-    down* {.bitsize:1.}: bool
-    r* {.bitsize:1.}: bool
-    l* {.bitsize:1.}: bool
-    notUsed* {.bitsize:6.}: cuint
+  KeypadRegs = KeyInputs | KEYCNT
+
+  KeyInput* = enum
```

Comment:
The move to enum sets seems a little odd to me. I/O registers in the GBA are fundamentally bitfields, and representing _some_ of them to be enum sets rather than bitfields because the fields happen to be flags doesn't make a ton of sense to me

Reply:
Enum sets are just better bitfields. though that's up to you to decide.